### PR TITLE
feat: Redirect the user to purchase overview and prefill fields with the previously purchased ticket

### PR DESCRIPTION
**Pull request by jorelosorio**
Closes https://github.com/AtB-AS/kundevendt/issues/1774

### CODE REVIEW

File: src/screens/Ticketing/Purchase/Overview/index.tsx [hunk 1]
```diff
@@ -82,19 +82,33 @@ const PurchaseOverview: React.FC<OverviewProps> = ({
...
+    defaultUserProfilesWithCount.forEach((u) => (u.count = 0));
+    params.userProfilesWithCount?.forEach((u: UserProfileWithCount) => {
+      defaultUserProfilesWithCount
+        .filter(
+          (upc: UserProfileWithCount) => upc.userTypeString == u.userTypeString,
+        )
+        .forEach((v: UserProfileWithCount) => (v.count = u.count));
+    });
+  }
+
```
Comment (gorandalum) [inline src/screens/Ticketing/Purchase/Overview/index.tsx:111]: I think we can try to simplify this a little bit, also the logic in `useDefaultUserProfilesWithCount`. The logic should be something like this:
- Get the user profiles from Firestore
- Filter by the limitations on the preassigned fare product
- If userProfilesWithCount in screen parameter: Apply them
- If no userProfilesWithCount in screen parameter: Apply default
- Use this as the initial state for travellerSelection

**Review by gorandalum — CHANGES_REQUESTED** [review]
Good work, and everything looks correct! Just suggested some refactoring to make this more maintainable for future developers, as both the already existing `useDefaultUserProfilesWithCount` and the new `filter`-/`forEach`-logic is quite complex.

File: src/screens/Ticketing/Purchase/Overview/index.tsx (hunk 1 above)
Comment (jorelosorio) [reply src/screens/Ticketing/Purchase/Overview/index.tsx]: I improved this

Comment (gorandalum) [reply src/screens/Ticketing/Purchase/Overview/index.tsx]: Slightly better, although still kinda complex, but haven't seen into if it actually possible to do in a less complex manner. Approving now!

**Review by gorandalum — APPROVED** [review]

File: src/screens/Ticketing/Purchase/Overview/index.tsx (hunk 1 above)
Comment (jorelosorio) [reply src/screens/Ticketing/Purchase/Overview/index.tsx]: I agree it is complex, I did the easiest possible with the current code solution. I have also changed the naming of some vars for a better understanding of what is going on.

**Pull request merged**